exaudfclient/docs/script_options_requirments.md

```diff
@@ -0,0 +1,313 @@
+| vertical tab | '\v'          | 11         | 0x0b     |
+| form feed    | '\f'          | 12         | 0x0c     |
+| space        | ' '           | 30        | 0x20      |     
+|======================================================= 
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+
+### Leading White Spaces Options Parsing
+`req~leading-white-spaces-script-options-parsing~1`
+
+The parser must recognize Script Options for lines starting with white space characters. 
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+Depends:
+ - `req~white-spaces~1`
+
+### Ignore anything which is not a Script Option
+`req~ignore-none-script-options~1`
+
+If there is any character in front of a Script Option which is not a white space, the parser must ignore the option(s). 
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+Depends:
+ - `req~white-spaces~1`
+
+### Multiple Line Script Options Parsing
+`req~multiple-lines-script-options-parsing~1`
+
+The parser must recognize Script Options at any line in the given script code.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+### White Spaces Options Parsing V1
+`req~white-spaces-script-options-parsing-v1~1`
+
+All white spaces between the option key and option value are to be interpreted as separator. 
+White spaces between the option value and the terminating ";" are to be removed from the option value.
+
+Needs: dsn
+
+Tags: V1
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+Depends:
+ - `req~white-spaces~1`
+
+### White Spaces Options Parsing V2
+`req~white-spaces-script-options-parsing-v2~1`
+
+All white spaces between the option key and option value are to be ignored. The following rules for escape sequences at **the start** of a script optionValue are to be applied:
+- '\ ' => space character
+- '\t' => <tab> character
+- '\f' => <form feed> character
+- '\v' => <vertical tab> character
+
+White spaces between the option value and the terminating ";" shall remain.
+
+Needs: dsn
+
+Tags: V2
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+Depends:
+ - `req~white-spaces~1`
+
+### Multiple and Options
+`req~multiple-options-management~1`
+
+The parser must collect multiple Script Options with the same key. Note: the specific handling depends on the option handler.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+### Duplicate Options Management
+`req~multiple-options-management~1`
+
+The parser must collect multiple Script Options with the same key and value. Note: the specific handling depends on the option handler.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+### Script Option Removal
+`req~script-option-removal~1`
+
+The parser handler must remove found known Script Options from the original script code. Successfully parsed Script Options, which are not recognized by the parser handler shall remain in the script code.
```

Comment:
Note that I have used here the term `parser handler` as it is not the parser itself which removes the found options from the source script code, but the client.

Reply:
This is not how the current parser works , if we want to change that we need to split the requirement per version and need to change the new implementation

Reply:
I extracted the second part:
```
Successfully parsed Script Options, which are not recognized by the parser handler shall remain in the script code.
```
into two new requirements: One for v1 (which behaves exactly as described here), and a new one for V2 which will throw an exception if an unknown option is detected (This is actually already implemented in that way in the new parser)